packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/index.tsx

```diff
@@ -130,6 +135,9 @@ const ConfirmSignupCode = ({
           scopes: integration.getPermissions(),
         }),
         ...(service !== MozServices.Default && { service }),
+        metricsContext: queryParamsToMetricsContext(
+          flowQueryParams as unknown as Record<string, string>
```

Comment:
Not here but I think we can do this in a nicer way... `flowQueryParams` is simply `window.location.search`, so passing it around into all our components and then recasting it here feels funny. I wonder if we should add `metricsContext` to the base integration so we can use `class-validator` to validate it and then we can just reference `integration.metricsContext()` or something when we need to call it.

Reply:
filed: https://mozilla-hub.atlassian.net/browse/FXA-9687